src/instruments/src/PFD/FMA.tsx

```diff
@@ -1,554 +1,1021 @@
+                this.cellsToHide.instance.style.visibility = 'hidden';
+            } else {
+                this.cellsToHide.instance.style.visibility = 'visible';
+            }
+        });
+    }
+
+    render(): VNode {
         return (
             <g>
-                <path className="NormalStroke Amber" d="m25.114 1.8143v13.506h-16.952v-13.506z" />
-                <text className="FontMedium MiddleAlign White" x="16.782249" y="7.1280665">MAN</text>
-                <text className="FontMedium MiddleAlign White" x="16.869141" y="14.351689">THR</text>
+                <g ref={this.cellsToHide}>
+                    <B2Cell bus={this.props.bus} />
+                    <C2Cell bus={this.props.bus} />
+                </g>
+                <E2Cell bus={this.props.bus} />
             </g>
         );
-    case 7:
-        text = 'SPEED';
-        break;
-    case 8:
-        text = 'MACH';
-        break;
-    case 9:
-        text = 'THR MCT';
-        break;
-    case 10:
-        text = 'THR CLB';
-        break;
-    case 11:
-        text = 'THR LVR';
-        break;
-    case 12:
-        text = 'THR IDLE';
-        break;
-    case 13:
+    }
+}
+
+class Row3 extends DisplayComponent<{ bus:EventBus, isAttExcessive: Subscribable<boolean> }> {
+    private cellsToHide = FSComponent.createRef<SVGGElement>();
+
+    onAfterRender(node: VNode): void {
+        super.onAfterRender(node);
+
+        this.props.isAttExcessive.sub((a) => {
+            if (a) {
+                this.cellsToHide.instance.style.visibility = 'hidden';
+            } else {
+                this.cellsToHide.instance.style.visibility = 'visible';
+            }
+        });
+    }
+
+    render(): VNode {
         return (
             <g>
-                <path className="NormalStroke Amber BlinkInfinite" d="m0.70556 1.8143h30.927v6.0476h-30.927z" />
-                <text className="FontMedium MiddleAlign Green" x="16.782249" y="7.1280665">A.FLOOR</text>
+                <A3Cell bus={this.props.bus} />
+                <g ref={this.cellsToHide}>
+                    <AB3Cell bus={this.props.bus} />
+                    <D3Cell bus={this.props.bus} />
+                </g>
+                <BC3Cell isAttExcessive={this.props.isAttExcessive} bus={this.props.bus} />
+                <E3Cell bus={this.props.bus} />
             </g>
         );
-    case 14:
+    }
+}
+
+interface CellProps extends ComponentProps {
+    bus: EventBus;
+}
+
+class A1A2Cell extends ShowForSecondsComponent<CellProps> {
+    private athrModeSub = Subject.create(0);
+
+    private cellRef = FSComponent.createRef<SVGGElement>();
+
+    constructor(props) {
+        super(props, 9);
+    }
+
+    onAfterRender(node: VNode): void {
+        super.onAfterRender(node);
+
+        const sub = this.props.bus.getSubscriber<PFDSimvars>();
+
+        sub.on('AThrMode').whenChanged().handle((athrMode) => {
+            this.athrModeSub.set(athrMode);
+
+            let text: string = '';
```

Comment:
I don't quite understand this part and would like to learn more about when this is appropriate. Why is this specific case better solved via strings? I don't feel like I see this pattern in a lot of other places in this PR.

Reply:
This is probably something we have to recheck in the future. Most other places in the PFD have a "static" structure where only the content or animations change. In the case of some Cells in the FMA the structure is very different dependent on the message so I was going for a "dumb" `.innerHtml` replace to properly render the content. 
I'd like to revisit this in the future. For example I could add all messages (or all that share a common element structure) and update/hide them accordingly. But that needs some trial and error on my side